app/src/main/java/com/zulip/android/gcm/GcmBroadcastReceiver.java

```diff
@@ -22,13 +29,30 @@
 public class GcmBroadcastReceiver extends BroadcastReceiver {
 
     private static final String TAG = "GCM";
-
+    public static final String CANCEL = "Cancel";
     public static String getGCMReceiverAction(Context appContext) {
         return appContext.getPackageName() + ".PushMessage.BROADCAST";
     }
 
     @Override
     public void onReceive(Context context, Intent intent) {
+        // handle cancel notification action on uploads
```

Comment:
Declaring the hashMap private and calling a function from here to zulipActivity would be better! 
Also add a check if `ZulipApp.get().getZulipActivity().hashMap;` returns null, as suppose if the app gets destroyed and notification is still hanging there then there would be crash!